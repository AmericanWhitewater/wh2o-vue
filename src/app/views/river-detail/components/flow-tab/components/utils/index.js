import { humanReadable } from "@/app/global/services";
import { formatReadingWithFormat } from "@/app/global/lib/gages";

export function formatReading(reading, metricID) {
  console.log(this.getMetric(metricID)?.format);
  return formatReadingWithFormat(
    reading,
    this.getMetric(metricID)?.format || ""
  );
}
export function getMetric(metricID) {
  if (metricID && this.metrics?.length) {
    return this.metrics.find((m) => m.id.toString() === metricID.toString());
  }
  return null;
}

export function formatTime(epoch) {
  return humanReadable(new Date().valueOf() - epoch * 1000);
}
export function formatFlowRange(min, max, metricID) {
  console.log(min, max, metricID);
  if (min && max) {
    return `${this.formatReading(min, metricID)} â€“ ${this.formatReading(
      max,
      metricID
    )} ${this.formatMetric(metricID)}`;
  }
  return "n/a";
}
export function formatMetric(metricId) {
  if (this.metrics) {
    return this.getMetric(metricId)?.unit || "n/a";
  }
  return "n/a";
}
