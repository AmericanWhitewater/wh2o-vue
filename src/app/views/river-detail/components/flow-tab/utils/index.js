export * from './gage-chart-config'
