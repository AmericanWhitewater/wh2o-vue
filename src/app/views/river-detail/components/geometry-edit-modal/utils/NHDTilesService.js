import bboxPolygon from '@turf/bbox-polygon'
import booleanPointInPolygon from '@turf/boolean-point-in-polygon'
import booleanPointOnLine from '@turf/boolean-point-on-line'
import lineIntersect from '@turf/line-intersect'
import lineSlice from '@turf/line-slice'
import pointToLineDistance from '@turf/point-to-line-distance'
import { lineString } from '@turf/helpers'
import Graph from 'graph-data-structure'

// this is a class so it can be initialized with the maplibregljs map object
// then continue to use it to populate and refresh the graph and lines caches
class NHDTilesService {
  constructor(map) {
    this.map = map;
    this.graphCache = {};
    this.linesCache = {};
  }

  calculateGeom(reachStart, reachEnd, currentGeom) {
    const graph = this.getNhdGraph()
    const lines = this.getNhdLines()


      // if the start and finish are not both visible in the current viewport,
      // we need to operate differently, so detect that here:
      const mapBbox = bboxPolygon(this.map.getBounds().toArray().flat())
      const intersection = lineIntersect(currentGeom, mapBbox)

      let newSegmentStart
      let newSegmentEnd
      let whichPointIsOutside
      let intersectionPoint
      // geom intersects viewport!
      if (intersection.features.length) {
        intersectionPoint = intersection.features[0]
        // determine whether reachStart or reachEnd is in the viewport
        if (booleanPointInPolygon(reachStart, mapBbox)) {
          whichPointIsOutside = 'end'
          newSegmentStart = reachStart
          newSegmentEnd = intersectionPoint
        } else {
          whichPointIsOutside = 'start'
          newSegmentStart = intersectionPoint
          newSegmentEnd = reachEnd
        }
      } else {
        newSegmentStart = reachStart
        newSegmentEnd = reachEnd
      }

      // use the segment start/end to generate a new path between those segments
      // turf has a bug with nearestPointOnLine:
      // https://github.com/Turfjs/turf/issues/1726
      // so we're identifying the line closest using pointToLineDistance instead
      const putinDistances = lines.map(x => pointToLineDistance(newSegmentStart, x));
      const putinSegment = lines[putinDistances.indexOf(Math.min(...putinDistances))];

      const takeoutDistances = lines.map(x => pointToLineDistance(newSegmentEnd, x));
      const takeoutSegment = lines[takeoutDistances.indexOf(Math.min(...takeoutDistances))];
      
      const shortestPath = graph.shortestPath(
        putinSegment.id.toString(),
        takeoutSegment.id.toString()
      )
      const coords = []
      shortestPath.forEach(x => {
        const l = lines.find(l => (l.id.toString() === x))
        coords.push(...l.geometry.coordinates)
      })
      const initialLine = lineString(coords)
      const slicedLine = lineSlice(newSegmentStart, newSegmentEnd, initialLine)

      let newGeom

      // if the segment extends beyond the viewport, merge it with the original geom, sliced at the viewport overlap
      if (whichPointIsOutside) {
        let fullCoords
        if (whichPointIsOutside === 'end') {
          const originalSegment = lineSlice(intersectionPoint, reachEnd, currentGeom)
          fullCoords = [...slicedLine.geometry.coordinates, ...originalSegment.geometry.coordinates]
        } else {
          const originalSegment = lineSlice(reachStart, intersectionPoint, currentGeom)
          fullCoords = [...originalSegment.geometry.coordinates, ...slicedLine.geometry.coordinates]
        }
        newGeom = lineString(fullCoords)
      } else {
        newGeom = slicedLine
      }
      return newGeom
  }

  getNhdLines () {
    const cacheKey = this.map.getBounds().toString()
    if (this.linesCache[cacheKey] == null) {
      this.linesCache = {}
      const mapFeatures = this.map.queryRenderedFeatures({ layers: ['nhdflowline'] })
      const lines = mapFeatures
        // mapFeatures sometimes has multiLineStrings in it that need to be split for our graph algorithm
        .flatMap(x => {
          if (x.geometry.type === 'MultiLineString') {
            return x.geometry.coordinates.map((coords, index) => (
              lineString(coords, x.properties, { id: `${x.id}-${index}`})
            ))
          } else {
            return x
          }
        })
        .filter(x => (
          x.geometry.type === 'LineString' &&
          // this is mostly taken from Seth's code, we may want to revisit?
          // ideally improves speed by excluding some NHD flowlines from graph
          ![31800, 33601, 33603, 34300, 34305, 34306, 36400, 40300, 40307, 40308, 40309, 44500, 46100, 48400, 48500, 56800].includes(x.properties.fcode)
        ))
      this.linesCache[cacheKey] = lines
    }
    return this.linesCache[cacheKey]
  }

  getNhdGraph () {
    const graphCacheKey = this.map.getBounds().toString()
    if (this.graphCache[graphCacheKey] == null) {
      this.graphCache = {}

      const graph = new Graph()
      const lines = this.getNhdLines()
      lines.forEach((x) => { graph.addNode(x.id.toString()) })

      lines.forEach((x, index) => {
        lines.slice(index + 1).forEach(y => {
          if (x !== y &&
              (booleanPointOnLine(y.geometry.coordinates[0], x) ||
               booleanPointOnLine(y.geometry.coordinates.slice(-1)[0], x))) {
            graph.addEdge(x.id.toString(), y.id.toString())
            graph.addEdge(y.id.toString(), x.id.toString())
          }
        })
      })
      this.graphCache[graphCacheKey] = graph
    }
    return this.graphCache[graphCacheKey]
  }
}

export { NHDTilesService };