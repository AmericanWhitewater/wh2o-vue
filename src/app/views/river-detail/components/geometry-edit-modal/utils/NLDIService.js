import lineSlice from '@turf/line-slice'
import { lineString } from '@turf/helpers'


// most of the NLDI logic here is reproduced from
// https://github.com/sdl60660/river-runner

export default {
  async calculateGeom(reachStart, reachEnd) {
    let slicedReach;
    // attempt to use the NLDI to get the flowline matching the requested points
    try {
      let closestFeature;
      closestFeature = await this.findClosestFeature(reachStart);
      let flowlinesData;
      flowlinesData = await this.fetchNLDI(closestFeature);
      const combinedFlowlines = flowlinesData.features[0];
      combinedFlowlines.geometry.coordinates = flowlinesData.features
        .map((a) => a.geometry.coordinates)
        .flat();
      slicedReach = lineSlice(reachStart, reachEnd, combinedFlowlines);
    } catch {
      // if NLDI doesn't work for whatever reason, just return a straight line
      slicedReach = lineString([
        reachStart.geometry.coordinates,
        reachEnd.geometry.coordinates,
      ]);
    }
    return slicedReach;
  },
  async findClosestFeature(point) {
    let closestFeature;
    let resultFound = false;
    let roundingDigits = 6;
    while (resultFound === false && roundingDigits >= 0) {
      roundingDigits -= 1;
      const closestFeatureURL = `https://labs.waterdata.usgs.gov/api/nldi/linked-data/comid/position?coords=POINT%28${point.geometry.coordinates[0].toFixed(
        roundingDigits
      )}%20${point.geometry.coordinates[1].toFixed(roundingDigits)}%29`;
      const coordinateResponse = await fetch(closestFeatureURL);
      const data = await coordinateResponse.json();
      closestFeature = data.features[0];
      resultFound = true;
    }
    return closestFeature;
  },
  async fetchNLDI(closestFeature) {
    const siteURL =
      closestFeature.properties.navigation +
      "/DM/flowlines" +
      "?f=json&distance=6000";
    const response = await fetch(siteURL);
    try {
      const data = await response.json();
      return data;
    } catch {
      return null;
    }
  },
};