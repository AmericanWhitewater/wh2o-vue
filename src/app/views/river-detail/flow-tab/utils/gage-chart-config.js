export const GageChartConfig = {
  data: () => ({
    chartHeight: 200,
    chartConfig: {
      legend: {
        display: false
      },
      elements: {
        line: {
          tension: 0, // disables bezier curves
          borderWidth: 2,
          color: '#152935'
        },
        point: {
          radius: 1,
          hitRadius: 16,
          borderColor: 'rgba(0,0,0,0)',
          hoverBackgroundColor: '#152935'
        }
      },
      layout: {
        padding: 0
      },
      responsive: true,
      hover: {
        animationDuration: 300 // duration of animations when hovering an item
      },
      responsiveAnimationDuration: 0, // animation duration after a resize
      tooltips: {
        backgroundColor: '#ffffff',
        bodyFontColor: '#152934',
        bodyFontFamily: "'IBM Plex Sans' , sans-serif",
        bodyFontSize: 22,
        bodyLineHeight: 25,
        bodyFontStyle: 400,
        borderColor: '#DFE3E6',
        borderWidth: 1,
        caretSize: 8,
        cornerRadius: 0,
        displayColors: false,
        enabled: true,
        titleFontColor: '#152934',
        titleFontFamily: "'IBM Plex Sans' , sans-serif",
        titleFontSize: 12,
        titleLineHeight: 15,
        titleFontStyle: 600,
        titleMarginBottom: 4,
        xPadding: 16,
        yPadding: 16
      },
      scales: {
        xAxes: [
          {
            type: 'time',
            display: true,
            distribution: 'series',
            time: {
              unit: 'h:mm a',
              displayFormats: {
                day: 'h:mm a',
                week: 'll',
                month: 'll',
                year: 'MMM YYYY'
              }
            },
            gridLines: {
              color: 'rgba(90, 104, 114, 0.2)',
              borderDash: [4, 4]
            },
            scaleLabel: {
              display: true,
              labelString: 'Timespan',
              fontFamily: "'IBM Plex Sans' , 'sans-serif'",
              fontSize: 14
            },
            bounds: 'data',
            ticks: {
              source: 'data',
              beginAtZero: false,
              autoSkip: false,
              maxRotation: 45,
              minRotation: 45,
              // labelOffset: 20,
              fontFamily: "'IBM Plex Sans' , 'sans-serif'",
              fontSize: 13,
              min: 7
            }
          }
        ],
        yAxes: [
          {
            scaleLabel: {
              display: true,
              labelString: 'Reading',
              bounds: 'data',
              fontFamily: "'IBM Plex Sans' , 'sans-serif'",
              fontSize: 14
            },
            gridLines: {
              color: 'rgba(90, 104, 114, 0.2)',
              borderDash: [4, 4]
            },
            ticks: {
              beginAtZero: true,
              // suggestedMax: yMax,
              fontFamily: "'IBM Plex Sans' , 'sans-serif'"
              // fontSize: this.yTickFontSize
            }
            // beforeBuildTicks: axis => {
            //   opts.scales.yAxes[0].ticks["min"] = 100;
            // },
            // afterBuildTicks: function(axis) {
            //   // if (axis.ticks.length > 2) {
            //   //     axis.ticks.unshift(axis.ticks[0] * 1.3);
            //   //    // console.log(axis.ticks);
            //   // }
            // }
          }
        ]
      }
    }
  })
  // computed: {
  //   setYMax () {
  //     let yMax = this.$store.state.riverDetailState.gageReadingsData.data[0];
  //     console.log(yMax)
  //     yMax = Math.ceil(yMax * 1.3);
  //     this.chartConfig.scales.yAxes.ticks.suggestedMax = yMax
  //     return yMax
  //   }
  // }
}
