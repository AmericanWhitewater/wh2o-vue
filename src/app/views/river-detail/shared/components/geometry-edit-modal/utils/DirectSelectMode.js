import MapboxDraw from '@mapbox/mapbox-gl-draw'
import Constants from '@mapbox/mapbox-gl-draw/src/constants'
import { isOfMetaType } from '@mapbox/mapbox-gl-draw/src/lib/common_selectors'

const isVertex = isOfMetaType(Constants.meta.VERTEX)

export default {
  ...MapboxDraw.modes.direct_select,
  dragFeature (state, e, delta) {},
  onMouseMove (state, e) {
    // On mousemove that is not a drag, stop vertex movement.
    const onVertex = isVertex(e)
    const noCoords = state.selectedCoordPaths.length === 0
    // if (isFeature && noCoords) this.updateUIClasses({ mouse: Constants.cursors.MOVE });
    if (onVertex && !noCoords) this.updateUIClasses({ mouse: Constants.cursors.MOVE })
    else this.updateUIClasses({ mouse: Constants.cursors.NONE })
    this.stopDragging(state)

    // Skip render
    return true
  },
  // prevent it from leaving direct_select mode
  clickNoTarget () {},
  clickInactive () {}
}
