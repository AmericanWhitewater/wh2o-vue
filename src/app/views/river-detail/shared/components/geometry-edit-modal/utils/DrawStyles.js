import styles from '@mapbox/mapbox-gl-draw/src/lib/theme'

export default [
  // include default mapbox draw styleset
  ...styles,
  {
    id: 'basic-reach-line',
    type: 'line',
    filter: ['all',
      ['==', 'active', 'false'],
      ['==', '$type', 'LineString']
    ],
    layout: {
      'line-cap': 'round',
      'line-join': 'round'
    },
    paint: {
      'line-color': 'HSL(192, 61%, 35%)',
      'line-width': ['interpolate',
        ['linear'],
        ['zoom'],
        11, 3,
        15, 5
      ],
      'line-opacity': ['interpolate',
        ['linear'],
        ['zoom'],
        11, 1,
        15, 0.6
      ]
    }
  },
  {
    id: 'reach-start-style',
    type: 'circle',
    filter: ['all',
      ['==', '$type', 'Point'],
      ['==', 'meta', 'feature'],
      ['==', 'active', 'false'],
      ['==', 'id', 'reachStart']],
    paint: {
      'circle-radius': 7,
      'circle-color': 'hsl(120, 100%, 25%)',
      'circle-stroke-color': 'white',
      'circle-stroke-width': 0.75
    }
  },
  {
    id: 'reach-end-style',
    type: 'circle',
    filter: ['all',
      ['==', '$type', 'Point'],
      ['==', 'meta', 'feature'],
      ['==', 'active', 'false'],
      ['==', 'id', 'reachEnd']],
    paint: {
      'circle-radius': 7,
      'circle-color': 'hsl(0, 83%, 57%)',
      'circle-stroke-color': 'hsl(0, 0%, 100%)',
      'circle-stroke-width': 0.75
    }
  },
  {
    id: 'active-reach-start-style',
    type: 'circle',
    filter: ['all',
      ['==', '$type', 'Point'],
      ['==', 'meta', 'feature'],
      ['==', 'active', 'true'],
      ['==', 'id', 'reachStart']],
    paint: {
      'circle-radius': 5,
      'circle-color': 'hsl(120, 100%, 30%)',
      'circle-stroke-color': '#fbb03b',
      'circle-stroke-width': 2.5
    }
  },
  {
    id: 'active-reach-end-style',
    type: 'circle',
    filter: ['all',
      ['==', '$type', 'Point'],
      ['==', 'meta', 'feature'],
      ['==', 'active', 'true'],
      ['==', 'id', 'reachEnd']],
    paint: {
      'circle-radius': 5,
      'circle-color': 'hsl(0, 83%, 67%)',
      'circle-stroke-color': '#fbb03b',
      'circle-stroke-width': 2.5
    }
  }
]
