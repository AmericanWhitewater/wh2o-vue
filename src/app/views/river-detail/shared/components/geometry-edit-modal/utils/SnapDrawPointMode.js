import MapboxDraw from '@mapbox/mapbox-gl-draw'
import Constants from '@mapbox/mapbox-gl-draw/src/constants'

import knn from 'rbush-knn'
import polygonToLine from '@turf/polygon-to-line'
import rbush from 'geojson-rbush'
import nearestPointOnLine from '@turf/nearest-point-on-line'

let cache = {}

export default {
  ...MapboxDraw.modes.draw_point,
  onClick (state, e) {
    this.updateUIClasses({ mouse: Constants.cursors.MOVE })
    const snappedPoint = this._getSnappedPoint(e.lngLat)
    state.point.updateCoordinate('', ...snappedPoint)
    this.map.fire(Constants.events.CREATE, {
      features: [state.point.toGeoJSON()]
    })
  },
  _getSnappedPoint (target) {
    const candidates = this._getCandidates()
    const nearestLines = knn(candidates, target.lng, target.lat, 10)

    if (nearestLines.length === 0) {
      return
    }

    const nearest = nearestLines
      .map(l => [l, nearestPointOnLine(l, [target.lng, target.lat])])
      .sort((a, b) => a[1].properties.dist - b[1].properties.dist)
      .shift()

    return nearest[1].geometry.coordinates
  },
  _getCandidates () {
    const key = this.map.getBounds().toString()

    if (cache[key] == null) {
      // clear cache
      cache = {}

      const candidates = this.map
        // querySourceFeatures("composite", { sourceLayer: "nhdpolygon" }) also
        // works, but there's something nice about syncing what's visible with
        // where points can snap
        .queryRenderedFeatures({ layers: this.config.layers })
        .map(x => {
          if (
            [
              Constants.geojsonTypes.POLYGON,
              Constants.geojsonTypes.MULTI_POLYGON
            ].includes(x.geometry.type)
          ) {
            return polygonToLine(x, x.properties)
          }

          return x
        })
        .flatMap(x => {
          if (x.type === Constants.geojsonTypes.FEATURE_COLLECTION) {
            return x.features
          }

          return x
        })

      if (candidates.length === 0) {
        // nowhere to snap to
        return null
      }

      cache[key] = rbush().load(candidates)
    }

    return cache[key]
  }
}
