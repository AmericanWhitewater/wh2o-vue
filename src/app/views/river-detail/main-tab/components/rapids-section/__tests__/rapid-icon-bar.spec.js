import { createWrapper } from '@/utils'
import RapidIconBar from '../components/rapid-icon-bar.vue'

const mockStore = {
  state: {
    appGlobalState: {
      appGlobalData: {
        editMode: null
      }
    }
  }
}

const options = {
  mocks: {
    $store: mockStore
  },
  propsData: {
    character: null
  },
  stubs: ['WarningSquare32', 'Login32', 'Repeat32', 'Pedestrian32', 'Logout32']
}

describe('rapid-icon-bar.vue', () => {
  it('shows put in icon when poi is put in', () => {
    options.propsData.character = ['putin']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#put-in-icon').exists()).toBe(true)
  })
  it('shows access point icon when poi is access point', () => {
    options.propsData.character = ['access']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#access-icon').exists()).toBe(true)
  })
  it('shows hazard icon when poi is hazard', () => {
    options.propsData.character = ['hazard']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#hazard-icon').exists()).toBe(true)
  })
  it('shows surf icon when poi is playspot', () => {
    options.propsData.character = ['playspot']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#surf-icon').exists()).toBe(true)
  })
  it('shows portage icon when poi is portage', () => {
    options.propsData.character = ['portage']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#portage-icon').exists()).toBe(true)
  })
  it('shows waterfall icon when poi is waterfall', () => {
    options.propsData.character = ['waterfall']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#waterfall-icon').exists()).toBe(true)
  })
  it('shows take out icon when poi is take out', () => {
    options.propsData.character = ['takeout']
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#take-out-icon').exists()).toBe(true)
  })

  it('hides character icons and shows edit actions when edit mode enabled', () => {
    mockStore.state.appGlobalState.appGlobalData.editMode = true
    const wrapper = createWrapper(RapidIconBar, options)
    expect(wrapper.find('.rapid-icon-bar')).toMatchSnapshot()
    expect(wrapper.find('#character-icons-wrapper').exists()).toBe(false)
  })

  // maybe these emit tests arent needed?
  it('initiates edit process when edit button clicked', async () => {
    const wrapper = createWrapper(RapidIconBar, options)
    wrapper.find('#edit-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted('rapid:edit')).toBeTruthy()
  })
  it('initiates delete process when edit button clicked', async () => {
    const wrapper = createWrapper(RapidIconBar, options)
    wrapper.find('#delete-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted('rapid:delete')).toBeTruthy()
  })
})
