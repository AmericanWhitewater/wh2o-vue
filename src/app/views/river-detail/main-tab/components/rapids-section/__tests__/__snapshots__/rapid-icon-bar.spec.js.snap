// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rapid-icon-bar.vue hides character icons and shows edit actions when edit mode enabled 1`] = `
<div class="has-icons rapid-icon-bar">
  <div><button role="button" class="cv-button bx--btn bx--btn--secondary bx--btn--sm" id="edit-button">
      Edit
      <!---->
      <!----></button> <button role="button" class="cv-button bx--btn bx--btn--danger bx--btn--sm" id="delete-button">
      Delete
      <!---->
      <!----></button></div>
</div>
`;

exports[`rapid-icon-bar.vue shows access point icon when poi is access point 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <li id="access-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Access Point</span>
        <repeat32-stub></repeat32-stub></button></li>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows hazard icon when poi is hazard 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <!---->
    <li id="hazard-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Hazard</span>
        <warningsquare32-stub></warningsquare32-stub></button></li>
    <!---->
    <!---->
    <!---->
    <!---->
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows portage icon when poi is portage 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <!---->
    <!---->
    <!---->
    <li id="portage-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Portage</span>
        <pedestrian32-stub></pedestrian32-stub></button></li>
    <!---->
    <!---->
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows put in icon when poi is put in 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <li id="put-in-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Put In</span>
        <login32-stub></login32-stub></button></li>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows surf icon when poi is playspot 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <!---->
    <!---->
    <li id="surf-icon"><span><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Surf Spot</span><i class="app-icon"><svg id="surf-icon" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
          <!---->
          <g stroke-linecap="square" stroke-linejoin="miter" stroke-width="2" fill="#565656" stroke="#565656">
            <path d="M1,27a9.731,9.731,0,0,0,7.5-3.5,9.786,9.786,0,0,0,15,0A9.676,9.676,0,0,0,31,27" fill="none" stroke="#565656" stroke-miterlimit="10"></path>
            <path d="M1,19H2.364c8.181,0,6.42-8.74,11.612-13.075,5.7-4.758,10.855.677,11.664,1.693C19.976,8.634,21.455,19,29.636,19H31" fill="none" stroke-miterlimit="10"></path>
          </g>
        </svg></i></button></span></li>
    <!---->
    <!---->
    <!---->
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows take out icon when poi is take out 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li id="take-out-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Take Out</span>
        <logout32-stub></logout32-stub></button></li>
  </ul>
</div>
`;

exports[`rapid-icon-bar.vue shows waterfall icon when poi is waterfall 1`] = `
<div class="has-icons rapid-icon-bar">
  <ul id="character-icons-wrapper">
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li id="waterfall-icon"><button type="button" class="cv-tooltip bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"><span class="bx--assistive-text">Waterfall / Large Drop</span>
        <warningsquare32-stub></warningsquare32-stub></button></li>
    <!---->
  </ul>
</div>
`;
